### ISSUE TITLE
enable more metric profile options

### ISSUE THREAD

**Issue by alanzhai219** [huggingface/optimum-intel#337]
cant disable use_cache option from high level api.
It is hard code here.
https://github.com/huggingface/optimum-intel/blob/c8cec30474fb9c610991236fa0d16f5c609d1072/optimum/intel/openvino/modeling_decoder.py#L407
If the dynamic shape is not at index 3, this will report error because cant call a get_shape() from a dynamic tensor.

**Comment by echarlaix** [comment]
Hi @alanzhai219,

You can define this parameter when exporting a model : 

```python
model = OVModelForCausalLM.from_pretrained(model_id, export=True, use_cache=False)
```

This parameter will have an impact on the way your model is being exported. When loading an OpenVINO model, this attribute is being determenied by whether the `past_key_values` are defined as model inputs or not.


https://github.com/huggingface/optimum-intel/blob/c8cec30474fb9c610991236fa0d16f5c609d1072/optimum/intel/openvino/modeling_decoder.py#L96

**Comment by alanzhai219** [comment]
@echarlaix 
Thanks. BTW, is any possibility to enable and collect compile and infer time?

**Comment by echarlaix** [comment]
By default the model will be compiled when instantiating an `OVModel`. You can disable compilation by setting `compile=False` and trigger it by calling `.compile()`

```python
model = OVModelForXxxx.from_pretrained(...., compile=False)
model.compile()
```
If you don't call `.compile()`, compilation will be done before the first inference so we recomend to do it beforehand. For more information, you can check out our [documentation](https://huggingface.co/docs/optimum/main/en/intel/inference).

**Comment by alanzhai219** [comment]
@echarlaix 
Thanks. That is great. 
Could you add a new feature to measure each infer time? It will help analysis and improve the openvino performance.
